bin/fin

```diff
@@ -1471,31 +1477,44 @@ docker_machine_start ()
 			docker_machine_mounts
 	else
 		# only auto-create default machine
+		# TODO: Why do we need this?
 		if [[ "$DOCKER_MACHINE_NAME" != "$DEFAULT_MACHINE_NAME" ]]; then
 			echo-error "Machine '$DOCKER_MACHINE_NAME' does not exist"
 			echo-yellow "Removing reference to non-existent machine... Run your command again."
+			# TODO: this can be replaced with "docker-machine rm --force -y $DOCKER_MACHINE_NAME"
 			rm -f "$CONFIG_MACHINE_ACTIVE"
 			exit 1
 		fi
-		docker_machine_create && \
+
+		[[ $offline ]] && mode="--offline"
+
+		docker_machine_create "$DEFAULT_MACHINE_NAME" "$DEFAULT_MACHINE_PROVIDER" "$mode" && \
 			# Disable IP change for now
 			#docker_machine_change_ip && \
 			docker_machine_env
-
-		if [ $? -eq 0 ]; then
+		[[ $? != 0 ]] && _docker_machine_start_exception
+
+		# Load system images from docksal-system-images.tar (in the current directory) when provisioning in offline mode
+		if [[ $offline ]]; then
```

Comment:
Propose: reverse the if clause so that regular case was first and offline case was second.